platform/log/zephyr/src/mender-log.c

```diff
@@ -33,36 +33,47 @@ mender_log_init(void) {
     return MENDER_OK;
 }
 
+/* IMPORTANT: Please note that the default size of internal Zephyr logger buffer is 1024 bytes; if we need log buffer size 
+              to be more than that there is CONFIG_LOG_BUFFER_SIZE option allowing to extend the buffer size. */
+#define LOG_BUFFER_SIZE_BYTES (128)
+
 mender_err_t
 mender_log_print(uint8_t level, const char *filename, const char *function, int line, char *format, ...) {
 
+    (void)filename;
     (void)function;
-    char log[256] = { 0 };
 
-    /* Format message */
     va_list args;
     va_start(args, format);
-    vsnprintf(log, sizeof(log), format, args);
-    va_end(args);
+
+    char log_buff[LOG_BUFFER_SIZE_BYTES];
+    int  ret = vsnprintf(log_buff, LOG_BUFFER_SIZE_BYTES, format, args);
+
+    if (ret < 0) {
+        LOG_ERR("logger error: log message formatting failed: %d", ret);
+        va_end(args);
+        return MENDER_FAIL;
+    }
 
     /* Switch depending log level */
     switch (level) {
         case MENDER_LOG_LEVEL_ERR:
-            LOG_ERR("%s (%d): %s", filename, line, log);
+            LOG_ERR("(%d): %s", line, log_buff);
```

Comment:
To be honest, having the line with no filename misses the point. To save buffer size we could trim the filename to its basename. Otherwise I vote for using the function and no filename/line.

Reply:
Makes sense. Let's get rid of line as well maybe. Doing some extra operation in logging, like getting basename prom the filename will make it even more heavy. I don't like already that we need to process arguments in the log function.

Reply:
We should have filename and line for at least the debug messages. Right?

Reply:
Yup! Those messages should have it.

Reply:
I think function name (and line number) is enough.

The issue with filename is that it resolves to something like `WEST_TOPDIR/modules/mender-mcu/core/src/mender-client.c (610)` - that is why I suggested to get the basename only. But this complicates the code, so I support removing it in favor of the function name and line.